django/db/migrations/executor.py

```diff
@@ -288,9 +289,18 @@ def detect_soft_applied(self, project_state, migration):
                     continue
 
                 table = model._meta.db_table
-                db_field = model._meta.get_field(operation.name).column
+                field = model._meta.get_field(operation.name)
+
+                # Handle implicit many-to-many tables created by AddField.
+                if field.many_to_many:
+                    if field.remote_field.through._meta.db_table not in existing_table_names:
+                        return False, project_state
+                    else:
+                        found_add_field_migration = True
+                        continue
+
                 fields = self.connection.introspection.get_table_description(self.connection.cursor(), table)
-                if db_field not in (f.name for f in fields):
+                if field.column not in (f.name for f in fields):
```

Comment:
I guess renaming the `fields` variable to `columns` wouldn't hurt here. At first I assumed `fields` was a list of `db.models.Field` and thought there could be an issue with the use of `f.name` instead of `f.column`